deploy/tests/conftest.py

```diff
@@ -83,3 +84,15 @@ def patch(self, task_name: str, wrap_task=False, wrap_config=False) -> firesim.T
             return t
 
     return TaskMocker(mocker)
+
+
+@pytest.fixture
+def mock_paramiko_ssh_and_key():
```

Comment:
Might be better to name this
```suggestion
def mock_ssh_server():
    """Use paramiko to provide an ssh server on localhost
    
    Returns:
        server: mockssh.Server object with sample-user defined as a user
        sample_key: path to RSA key 'sample-user' can use to connect to ``server``
```

Reply:
Yeah it's a bit cleaner. That is what I originally named it but then added the _and_key_ because the return value contains a key too. However, I don't have a preference. If you make this change, make sure to change the fixture invocation in the test itself so it doesn't break.